src/ConnectedMode/Persistence/ServerConnectionsRepository.cs

```diff
@@ -0,0 +1,207 @@
+        if (wasDeleted && removedConnection != null)
+        {
+            credentialsLoader.DeleteCredentials(removedConnection.CredentialsUri);
+        }
+        return wasDeleted;
+    }
+
+    public bool TryUpdateSettingsById(string connectionId, ServerConnectionSettings connectionSettings)
+    {
+        return SafeUpdateConnectionsFile(() => TryUpdateConnectionSettings(connectionId, connectionSettings));
+    }
+
+    public bool TryUpdateCredentialsById(string connectionId, ICredentials credentials)
+    {
+        try
+        {
+            var wasFound = TryGet(connectionId, out ServerConnection serverConnection);
+            if (!wasFound)
+            {
+                return false;
+            }
+            credentialsLoader.Save(credentials, serverConnection.CredentialsUri);
+            return true;
+        }
+        catch (Exception ex)
+        {
+            logger.WriteLine(ex.Message);
+            return false;
+        }
+    }
+
+    private bool TryUpdateConnectionSettings(string connectionId, ServerConnectionSettings connectionSettings)
+    {
+        var wasFound = ServerConnectionsCache.TryGetValue(connectionId, out ServerConnection existingServerConnection);
+        if (!wasFound)
+        {
+            return false;
+        }
+
+        existingServerConnection.Settings = connectionSettings;
+        return true;
+    }
+
+    private bool SafeUpdateConnectionsFile(Func<bool> updateConnectionsCache)
+    {
+        try
+        {
+            ReadServerConnectionsFile();
+            ServerConnectionsCache ??= new ConcurrentDictionary<string, ServerConnection>();
+
+            var wasUpdated = updateConnectionsCache();
+            if (!wasUpdated)
+            {
+                return false;
+            }
+
+            var model = serverConnectionModelMapper.GetServerConnectionsListJsonModel(ServerConnectionsCache.Values);
+
+            return jsonFileHandle.TryWriteToFile(storageFilePath, model);
+        }
+        catch (Exception ex)
+        {
+            logger.WriteLine(ex.Message);
+            return false;
+        }
+    }
+
+    private static string GetStorageFilePath(IEnvironmentVariableProvider environmentVariables)
+    {
+        var appDataFolder = environmentVariables.GetSLVSAppDataRootPath();
+        return Path.Combine(appDataFolder, ConnectionsFileName);
+    }
+
+    private void ReadServerConnectionsFile()
+    {
+        try
+        {
+            if (!jsonFileHandle.TryReadFile(storageFilePath, out ServerConnectionsListJsonModel model))
+            {
+                return;
+            }
+
+            ServerConnectionsCache = new ConcurrentDictionary<string, ServerConnection>();
+            model.ServerConnections.ForEach(c => ServerConnectionsCache.TryAdd(c.Id, serverConnectionModelMapper.GetServerConnection(c)));
+        }
+        catch (Exception ex)
+        {
```

Comment:
This is interesting. Why are exceptions not handled here? Also, logically, if the TryReadFile never throws, there shouldn't be any exceptions here at all. Except for bugs, but If we shouldn't defend against that